src/common/base/predicate.ts

```diff
@@ -0,0 +1,11 @@
+/** A function that converts some input into a boolean, often use for calls like `filter`. */
+export type Predicate<T> = (input: T) => boolean;
+
+// static methods that hang off Predicate
+export namespace Predicate {
+  /** A Predicate that always succeeds */
+  export const alwaysTrue = <T>() => true;
+  /** A Predicate that always fails */
+  export const alwaysFalse = <T>() => false;
+  // TODO(lindner) and(..) negate(..) or(..) etc..
```

Comment:
Good idea.

Reply:
This pattern can be used for a lot of the other HOTs, rad.

Reply:
It's tricky because you can only tack-on static functions.  At some point a Preconditions class might need to exist to allow for chaining..   Here's some sample syntax:

```
  Collections.sort(array, Predicate.of((a,b)=>{...}).and(...).toFunction());
```